androidApp/src/main/java/co/nimblehq/avishek/phong/kmmic/android/ui/screen/surveydetail/SurveyDetailScreen.kt

```diff
@@ -1,32 +1,60 @@
 package co.nimblehq.avishek.phong.kmmic.android.ui.screen.surveydetail
 
+import androidx.compose.foundation.layout.fillMaxSize
 import androidx.compose.foundation.pager.*
 import androidx.compose.material.*
 import androidx.compose.runtime.*
+import androidx.compose.ui.Modifier
 import androidx.compose.ui.tooling.preview.Preview
 import androidx.compose.ui.tooling.preview.PreviewParameter
 import androidx.lifecycle.compose.collectAsStateWithLifecycle
 import co.nimblehq.avishek.phong.kmmic.android.ui.theme.ApplicationTheme
 import co.nimblehq.avishek.phong.kmmic.presentation.module.HomeViewModel
+import co.nimblehq.avishek.phong.kmmic.presentation.uimodel.*
 import org.koin.androidx.compose.getViewModel
 
 @Composable
 fun SurveyDetailScreen(
     homeViewModel: HomeViewModel,
     surveyId: String,
-    onBackClick: () -> Unit
+    onBackClick: () -> Unit,
 ) {
     val viewState = homeViewModel.viewState.collectAsStateWithLifecycle()
     val surveyUiModel = viewState.value.surveys.find { it.id == surveyId }
+    var surveyQuestionUiModels by remember { mutableStateOf<List<SurveyQuestionUiModel>>(emptyList()) }
```

Comment:
Please move the data logic to ViewModel 🙏

Reply:
I'd like to do it in the integrate task. Is that okay?

Reply:
For UI task, we can add mutable states inside the ViewModel right? That way I can see how the screen is going to fetch the data before the integration task and I can also see how the data is observed. I think we should implement it in UI task. 🙏

Reply:
Ideally, we update the VM and add unit tests in the same integrate PR. But I will update as you said. 🙇🏽